Remove deprecated calls for Jedi >= 0.16.0

bnavigator opened the pull request:
Further work on https://github.com/palantir/python-language-server/pull/775 which helped me for the OpenSUSE package (where Jedi has a version 55 commits after 0.16.0)

Foxboron committed:
Changed towards jedi 0.16

Signed-off-by: Morten Linderud <morten@linderud.pw>

dalthviz committed:
PR: Fix hover request for numpy alias (np) and ufuncs (#768)

* Fix hover request for numpy alias (np) and ufuncs

* Change hover test string

bnavigator committed:
remove deprecated params call

bnavigator committed:
remove deprecated names call

bnavigator committed:
Allow more references returned in test_references_builtin (#778)

Co-authored-by: Benjamin Greiner <greiner@dsi.uni-stuttgart.de>

ccordoba12 commented on pyls/plugins/jedi_completion.py:
```diff
@@ -142,9 +146,9 @@ def _format_completion(d, include_params=True):
-        positional_args = [param for param in d.params
+    sig = d.get_signatures()
+    if (include_params and sig and not is_exception_class(d.name)):
+        positional_args = [param for param in sig[0].params
```
Why are you taking here only the first signature? Also, could you provide an example of an object with multiple signatures?

bnavigator replied:
Because PyLS expects only one for completion, but the newer Jedi provides a list of signatures.
https://jedi.readthedocs.io/en/latest/docs/api.html#jedi.Script.get_signatures

I am pretty sure there are cases with multiple signatures for a completion.
https://jedi.readthedocs.io/en/latest/docs/api-classes.html#jedi.api.classes.BaseName.get_signatures
```
    def get_signatures(self):
        """
        Returns all potential signatures for a function or a class. Multiple
        signatures are typical if you use Python stubs with ``@overload``.

        :rtype: list of :class:`BaseSignature`
        """
        return [
            BaseSignature(self._inference_state, s)
            for s in self._get_signatures()
        ]
```

ccordoba12 commented on pyls/plugins/jedi_completion.py:
```diff
@@ -167,8 +171,9 @@ def _format_completion(d, include_params=True):
-        params = ', '.join([param.name for param in definition.params])
+    sig = definition.get_signatures()
+    if definition.type in ('function', 'method') and sig:
+        params = ', '.join([param.name for param in sig[0].params])
```
Also here, why `sig[0]`?

bnavigator replied:
see above

ccordoba12 commented on pyls/workspace.py:
```diff
@@ -213,16 +213,9 @@ def word_at_position(self, position):
-            source=self.source, path=self.path, all_scopes=all_scopes,
-            definitions=definitions, references=references, environment=environment,
-        )
+        script = self.jedi_script()
+        return script.get_names(all_scopes=all_scopes, definitions=definitions,
+                                references=references)
```
Why did you ditch `environment` here?

bnavigator replied:
Because `environment` is already pulled from the configuration in `self.jedi_script()` and is passed to the `jedi.Script()` constructor there. The `.get_names()` method does not have an `environment` parameter.

ccordoba12 reviewed (commented):
Thanks @bnavigator for your help! I left some comments for you.

bnavigator committed:
convert position into jedi line, column
fix path and project parameters for jedi_script
remove deprecated usages() calls

bnavigator commented:
Jedi 0.17.0 came out and bbfcce1 is an intermediate commit. This is still WIP, as stated in #744

bnavigator commented:
Rebased and completed. See https://github.com/palantir/python-language-server/issues/744#issuecomment-614294243

bnavigator committed:
jedi.Script keywords source vs code

source is deprecated in 0.17.0 in favor of code
but code did not exist in 0.16.0

bnavigator committed:
don't expect isabs at first place

sometimes jedi reports some keywords first

bnavigator committed:
check expected in all refs in builtin test

bnavigator commented:
I reused the same branch for #781 and after the rebase this does not merge into 755 anymore. closing.
